server/controllers/requests.js

```diff
@@ -0,0 +1,72 @@
+const mongoose = require("mongoose");
+const asycnHandler = require("express-async-handler");
+const Request = require("../models/Requests");
+const Requests = require("../models/Requests");
+const Profile = require("../models/Profile");
+
+// @route POST /request
+// Add a new request
+exports.newRequest = asycnHandler(async (req, res, next) => {
+    const request = req.body;
+    const newRequest = new Request({ ...request});
+
+    if(!newRequest) {
+        res.status(500);
+        throw new Error("Request could not be made.");
+    }
+    await newRequest.save();
+    res.status(200).json({ request: newRequest});
+});
+
+// @route GET /request
+// Get all users requests
+exports.userRequests = asycnHandler(async (req, res, next) => {
+    const id = req.params.user.id;
+    //find all the requests that have the users id, or sitter id
+    //this will include pending *and* completed requests, which can
+    //be handled by the front end, in order to display an accurate
+    //history, we can also group them in the app in case there are
+    //users who both are sitters, and use the app to find sitters
```

Comment:
Comments are nice, but we only really need to comment when it might be hard to understand something in the future or if you need to explain a trade-off to another developer working in the code base. This comment is rather large and not necessary as the code does a pretty good job of explaining itself.

Reply:
Okay! Thank you for the feedback!